### PR TITLE
Support unwrapping arrays for `JsonNode` decoder

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-serialization/issues/699

+ Added new methods to read input directly into the `JsonNode`

### CODE REVIEW

File: serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java [hunk 1]
```diff
@@ -35,12 +38,41 @@
...
+
+    /**
+     * Read value directly to as {@link JsonNode}.
+     * @param string The string
+     * @param type The type
+     * @return new {@link JsonNode} representing the JSON
+     * @throws IOException
+     * @since 2.9.0
+     */
+    default JsonNode readValueToTree(@NonNull String string, @NonNull Argument<?> type) throws IOException {
```
Comment (yawkat) [inline serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java:61]: readTree

File: serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java [hunk 2]
```diff
@@ -35,12 +38,41 @@
...
+
+    /**
+     * Read value directly to as {@link JsonNode}.
+     * @param is The input stream
+     * @param type The type
+     * @return new {@link JsonNode} representing the JSON
+     * @throws IOException
+     * @since 2.9.0
+     */
+    default JsonNode readValueToTree(@NonNull InputStream is, @NonNull Argument<?> type) throws IOException {
```
Comment (yawkat) [inline serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java:73]: what is the point of passing the `type`?

File: serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java [hunk 3]
```diff
@@ -35,12 +38,41 @@
...
-        return new String(writeValueAsBytes(object), StandardCharsets.UTF_8);
+    /**
+     * Read value directly to as {@link JsonNode}.
+     * @param byteArray The byte array
+     * @param type The type
+     * @return new {@link JsonNode} representing the JSON
+     * @throws IOException
+     * @since 2.9.0
+     */
+    default JsonNode readValueToTree(byte[] byteArray, @NonNull Argument<?> type) throws IOException {
```
[the remark this answers is not in the document]
Comment (dstepanov) [reply serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java:49]: Are you sure? We have `readValueFromTree` the oposite sounded fine :)

**Comment by yawkat** [comment]
Unwrapping LGTM.

I don't understand the point of the new tree methods. They convert bytes->tree, but you can already use `readValue(bytes, JsonNode.class)` for that. this seems to have an additional conversion step to an intermediate value, but i dont think it is necessary?

**Comment by dstepanov** [comment]
@yawkat I didn't realize it can be used that way, remove the new methods

**Comment by yawkat** [comment]
Maybe it makes sense to have aliases for it, but the impl can delegate to readValue(x, JsonNode) instead of going through another type.

File: serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (yawkat) [reply serde-api/src/main/java/io/micronaut/serde/ObjectMapper.java]: readTree is a jackson method that does the same.

**Review by graemerocher — APPROVED** [review]

**Review by yawkat — APPROVED** [review]

**Pull request merged**